scene/gui/code_edit.cpp

```diff
@@ -1291,15 +1291,22 @@ bool CodeEdit::is_line_numbers_zero_padded() const {
 
 void CodeEdit::_line_number_draw_callback(int p_line, int p_gutter, const Rect2 &p_region) {
 	String fc = TS->format_number(String::num(p_line + 1).lpad(line_number_digits, line_number_padding));
-	Ref<TextLine> tl;
-	tl.instantiate();
-	tl->add_string(fc, font, font_size);
-	int yofs = p_region.position.y + (get_line_height() - tl->get_size().y) / 2;
+	TextLine tl;
+	tl.add_string(fc, font, font_size);
+
+	if (p_line == 0) {
+		// cache line height
+		line_height = get_line_height();
```

Comment:
I don't think this is the correct fix, `TextEdit` should be calculating / caching `get_line_height` differently rather then forcing the caller to do it. It's the cause of slowdown in quite a lot of places. An idea is to keep track in `TextEdit::Text` as we also have the same issue with `get_max_width`.

Reply:
I agree, the real issue is; the code could be a lot better in general, when it came to the low level calls to the font lib its very crap performance in debug.

I had very bad results overall with changing here, with less hotspots in performance (spikes) but overall more cost. Decided to leave it out of the patch